src/sentry/rules/actions/base.py

```diff
@@ -83,56 +110,166 @@ def get_form_instance(self):
+                },
+            )
+            return
+        response = installation.create_issue(data)
+        issue_key_path = future.kwargs.get("issue_key_path")
+        issue_key = functools.reduce(operator.getitem, issue_key_path.split("."), response)
+        create_link(issue_key, integration, installation, event)
+
+
 class TicketEventAction(IntegrationEventAction):
     """Shared ticket actions"""
 
-    @property
-    def prompt(self):
-        return u"Create {}".format(self.ticket_type)
+    def __init__(self, *args, **kwargs):
+        super(IntegrationEventAction, self).__init__(*args, **kwargs)
+        integration_choices = [
+            (i.id, self.translate_integration(i)) for i in self.get_integrations()
+        ]
 
-    def generate_footer(self, rule_url):
-        raise NotImplementedError
+        if not self.get_integration_id() and integration_choices:
+            self.data[self.integration_key] = integration_choices[0][0]
+
+        self.form_fields = {
+            self.integration_key: {
+                "choices": integration_choices,
+                "initial": six.text_type(self.get_integration_id()),
+                "type": "choice",
+                "updatesForm": True,
+            }
+        }
+
+        dynamic_fields = self.get_dynamic_form_fields()
+        if dynamic_fields:
+            self.form_fields.update(dynamic_fields)
 
-    def build_description(self, event, installation):
+    def get_dynamic_form_fields(self):
         """
-        Format the description of the ticket/work item
+        Either get the dynamic form fields cached on the DB return `None`.
+
+        :return: (Option) Django form fields dictionary
         """
-        rule_url = u"/organizations/{}/alerts/rules/{}/{}/".format(
-            self.project.organization.slug, self.project.slug, self.rule.id
-        )
-        return installation.get_group_description(event.group, event) + self.generate_footer(
-            rule_url
-        )
+        form_fields = self.data.get("dynamic_form_fields")
+        if not form_fields:
+            return None
 
-    def _linked_issues(self, event, integration):
-        return ExternalIssue.objects.filter(
-            id__in=GroupLink.objects.filter(
-                project_id=self.project.id,
-                group_id=event.group.id,
-                linked_type=GroupLink.LinkedType.issue,
-            ).values_list("linked_id", flat=True),
-            integration_id=integration.id,
-        )
+        # Although this can be done with dict comprehension, looping for clarity.
+        if isinstance(form_fields, list):
+            fields = {}
+            for field in form_fields:
+                if "name" in field:
+                    fields[field["name"]] = field
+            return fields
+        return form_fields
+
+    def translate_integration(self, integration):
+        return integration.name
+
+    def fix_data_for_issue(self):
```

Comment:
Now that this is only called in jira/notify_action you can remove this from base.